scipy/optimize/_qap.py

```diff
@@ -0,0 +1,789 @@
+import numpy as np
+import operator
+from . import (linear_sum_assignment, minimize_scalar, OptimizeResult)
+from .optimize import _check_unknown_options
+
+from scipy._lib._util import check_random_state
+import itertools
+
+
+def quadratic_assignment(A, B, method="faq", options=None):
+    r"""
+    Solve the quadratic assignment problem.
+
+    This function solves the Quadratic Assignment Problem (QAP) and the
+    Graph Matching Problem (GMP).
+
+    Quadratic assignment solves problems of the following form:
+
+    .. math::
+
+        \min_P & \ {\ \text{trace}(A^T P B P^T)}\\
+        \mbox{s.t. } & {P \ \epsilon \ \mathcal{P}}\\
+
+    where :math:`\mathcal{P}` is the set of all permutation matrices,
+    and :math:`A` and :math:`B` are square matrices.
+
+    Graph matching tries to *maximize* the same objective function.
+    This algorithm can be thought of as finding the alignment of the
+    nodes of two graphs that minimizes the number of induced edge
+    disagreements, or, in the case of weighted graphs, the sum of squared
+    differences of edge weight disagreements.
```

Comment:
```suggestion
    disagreements, or, in the case of weighted graphs, the sum of squared
    edge weight differences.
```
Presumably not differences of disagreements.

Reply:
Done.